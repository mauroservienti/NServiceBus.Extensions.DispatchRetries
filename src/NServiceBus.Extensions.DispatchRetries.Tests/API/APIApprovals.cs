using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using PublicApiGenerator;
using System.Runtime.CompilerServices;

namespace NServiceBus.Extensions.DispatchRetries.Tests.API
{
    public class APIApprovals
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Approve_API()
        {
            var publicApi = ApiGenerator.GeneratePublicApi(typeof(DispatchRetriesEndpointConfigurationExtensions).Assembly, options:null);

            Approvals.Verify(publicApi);
        }
    }
}
