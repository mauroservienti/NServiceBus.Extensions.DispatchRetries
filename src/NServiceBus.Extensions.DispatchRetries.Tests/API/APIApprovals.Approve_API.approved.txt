[assembly: System.Reflection.AssemblyMetadata("RepositoryUrl", "https://github.com/mauroservienti/NServiceBus.Extensions.DispatchRetries")]
[assembly: System.Runtime.Versioning.TargetFramework(".NETCoreApp,Version=v8.0", FrameworkDisplayName=".NET 8.0")]
namespace NServiceBus
{
    public class DispatchRetriesConfiguration
    {
        public void DefaultBatchDispatchRetriesPolicy(Polly.AsyncPolicy batchDispatchRetryPolicy) { }
        public void DefaultBatchDispatchRetriesResilienceStrategy(Polly.ResiliencePipeline batchDispatchResiliencePipeline) { }
        public void DefaultImmediateDispatchRetriesPolicy(Polly.AsyncPolicy immediateDispatchRetryPolicy) { }
        public void DefaultImmediateDispatchRetriesResilienceStrategy(Polly.ResiliencePipeline immediateDispatchRetryResiliencePipeline) { }
    }
    public static class DispatchRetriesEndpointConfigurationExtensions
    {
        public static NServiceBus.DispatchRetriesConfiguration DispatchRetries(this NServiceBus.EndpointConfiguration configuration) { }
        public static NServiceBus.DispatchRetriesConfiguration DispatchRetries(this NServiceBus.EndpointConfiguration configuration, Polly.AsyncPolicy defaultBatchAndImmediateDispatchRetriesPolicy) { }
        public static NServiceBus.DispatchRetriesConfiguration DispatchRetries(this NServiceBus.EndpointConfiguration configuration, Polly.ResiliencePipeline defaultBatchAndImmediateDispatchRetriesResiliencePipeline) { }
    }
    public static class MessageHandlerContextExtensions
    {
        public static void OverrideBatchDispatchRetryPolicy(this NServiceBus.IMessageHandlerContext context, Polly.AsyncPolicy batchDispatchRetryPolicy) { }
        public static void OverrideBatchDispatchRetryResilienceStrategy(this NServiceBus.IMessageHandlerContext context, Polly.ResiliencePipeline batchDispatchRetryResiliencePipeline) { }
        public static void OverrideImmediateDispatchRetryPolicy(this NServiceBus.IMessageHandlerContext context, Polly.AsyncPolicy immediateDispatchRetryPolicy) { }
        public static void OverrideImmediateDispatchRetryResilienceStrategy(this NServiceBus.IMessageHandlerContext context, Polly.ResiliencePipeline immediateDispatchRetryResiliencePipeline) { }
    }
}